### PR TITLE
USWDS - Pages: Avoid reordering content in Documentation example

### LINKED ISSUE #4594: Documentation template tab order should match visual order
**Is your feature request related to a problem? Please describe.**

#### As a developer
 Who uses USWDS in my project, I expect that my generated stylesheets only includes styles relevant for my site, so that I'm not shipping unnecessary code to my users, prolonging their load times.

#### As a user

As a user who navigates a site using USWDS, I expect that load times are reasonable, so that I can access the content I need in a timely manner.

As a user, [I expect that I can tab through content in a logical order](https://www.w3.org/TR/WCAG20-TECHS/H4.html), so that there is not a disconnect between the visual and mechanical presentation of the content.

**Describe the solution you'd like**

Remove the file `src/stylesheets/components/_layout.scss`, or more specifically the `.usa-layout-docs__sidenav` and `.usa-layout-docs__main` style selectors.

Adjust [`components/layouts/docs-inner.njk`](https://github.com/uswds/uswds/blob/develop/src/components/layouts/docs-inner.njk) such that either (a) the placement of the side navigation is consistently before or after the main content, regardless of device size, or (b) the markup is duplicated and shown/hidden at the appropriate viewport size, but in such a way that ensures the [DOM order is always consistent with the visual order](https://www.w3.org/TR/WCAG20-TECHS/C27.html).

**Describe alternatives you've considered**

If USWDS needs styling for its own documentation, this should be compiled separate from what's included in USWDS.

**Additional context**

Related issue on `order` concerns: #4424

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

**Added mobile sidenav to the Documentation page template.** Sidenavs will now always follow logical HTML order on mobile and desktop views. Both navs use utility classes to change display settings based on viewport width.

## Breaking change

:warning: This is potentially breaking change.

### **Changed HTML**

This solution effectively duplicates the nav HTML and displays only the correctly placed in-page nav.

While this doesn’t affect the in-page nav component, we do offer this page as a [template](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-demo-5794/templates/documentation-page/) and the component code is changed as a result of this PR. _// Need to verify this causes a change_

### **Removed template classes:**

- `usa-layout-docs__main`: This is the class that had the styles that reordered element.
- `usa-layout-docs`: This selector unused in current styles.

## Related issue

Closes #4594

## Related pull requests

Alternate approach as discussed in #5681
[Changelog →](https://github.com/uswds/uswds-site/pull/2495) (Taken from #5681)

## Preview link

| Current (`develop`) | Feature branch |
|:-----|:-----|
| [Default] | [Preview] |
| [Template] (Site) | [Feature branch template] |

[Default]: https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/?path=/story/pages-documentation-page--documentation-page&p=all

[Preview]: https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-feature-doc-layout-order/?path=/story/pages-documentation-page--documentation-page

[Template]: https://designsystem.digital.gov/templates/documentation-page/

[Feature branch template]: https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-demo-5794/templates/documentation-page/

## Problem statement

In mobile, tab order doesn't match visual order in the documentation page example.

## Solution

Create a second, mobile only in-page nav. Use utility classes to dictate which is displayed.

## Testing and review

Visit [Documentation page →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-feature-doc-layout-order/?path=/story/pages-documentation-page--documentation-page)

### In desktop widths

1. Tab through navigation
2. Confirm tab order is correct
3. Confirm screen readers read nav before main content
4. Confirm second nav is hidden and not tabbed to or read by screen readers

### In mobile widths

1. Confirm nav is still at the bottom of the page
2. Confirm tab order is correct
3. Confirm screen readers read main content before nav
4. Confirm original nav is hidden and not tabbed to or read by screen readers

### Testing checklist

- [ ]  No visual regressions
- [ ]  Hidden navs are not targetted by tabbing
- [ ]  Hidden navs are not read by screen readers
- [ ] Screen readers read content in visual order
- [ ]  Tab order follows matches expectation.

### CODE REVIEW

**Review by amyleadem — CHANGES_REQUESTED** [review]
This is looking good. Added a suggestion to remove duplicate code. The suggestion will likely need to be tweaked and tested, but wanted to give you a starting point to let you know what I'm thinking. Let me know if you have questions!

**Comment by mahoneycm** [comment]
## Update 3/1/24

Loved @amyleadem's suggestion to implement the side-nav via `include`. Testing didn't show any regression and the component markdown output as expected. Tested via install on site.

File: packages/templates/usa-documentation/includes/_usa-docs-inner.twig [hunk 1]
```diff
@@ -1,55 +1,52 @@
+{% set settings = {
+  "aria_label": "Side navigation",
```
Comment (amyleadem) [inline packages/templates/usa-documentation/includes/_usa-docs-inner.twig:2]: Updating this to match current aria label
```suggestion
  "aria_label": "Secondary navigation",
```

File: packages/templates/usa-documentation/includes/_usa-docs-inner.twig [hunk 2]
```diff
@@ -1,55 +1,52 @@
...
+    },
+    {
+      "label": "Current page",
+      "current": true,
+      "subnav": [
+        {
+          "label": "Child link"
+        },
+        {
+          "label": "Child link",
```
Comment (amyleadem) [inline packages/templates/usa-documentation/includes/_usa-docs-inner.twig:16]: This is marked as current on the `develop` version
```suggestion
          "label": "Child link",
          "current": true,
```

**Review by amyleadem — CHANGES_REQUESTED** [review]
This is looking good! Just a couple small tweaks to the data, detailed in the comments below.

**Review by amyleadem — APPROVED** [review]
Looks good to me!

I tested the following:
- [x] Confirm no visual regressions from `develop`
- [x] Confirm that the HTML order matches the visual order
- [x] Confirm that the code meets standards

File: packages/templates/usa-documentation/includes/_usa-docs-inner.twig [hunk 3]
```diff
@@ -1,55 +1,53 @@
+{% set settings = {
```
Comment (mejiaj) [inline packages/templates/usa-documentation/includes/_usa-docs-inner.twig:1]: Renaming to `sidenav_settings` would make it immediately clear that it's a settings object for sidenav.

```suggestion
{% set sidenav_settings = {
```

File: packages/templates/usa-documentation/includes/_usa-docs-inner.twig [hunk 4]
```diff
@@ -1,55 +1,53 @@
...
-              </ul>
-            </li>
-            <li class="usa-sidenav__item">
-              <a href="">Parent link</a>
-            </li>
-          </ul>
-        </nav>
-
+      <div class="usa-layout-docs__sidenav display-none desktop:display-block desktop:grid-col-3">
+        {% include "@components/usa-sidenav/src/usa-sidenav.twig" with settings %}
```
Comment (mejiaj) [inline packages/templates/usa-documentation/includes/_usa-docs-inner.twig:47]: If you decide to update the variable name above.

```suggestion
        {% include "@components/usa-sidenav/src/usa-sidenav.twig" with sidenav_settings %}  
```

File: packages/templates/usa-documentation/includes/_usa-docs-inner.twig [hunk 5]
```diff
@@ -62,5 +60,10 @@
         <p>Read the full documentation on our side navigation on the component page.</p>
       </main>
     </div>
+      
+    <div class="usa-layout-docs__sidenav desktop:display-none">
+      {% include "@components/usa-sidenav/src/usa-sidenav.twig" with settings %}
```
Comment (mejiaj) [inline packages/templates/usa-documentation/includes/_usa-docs-inner.twig:65]:
```suggestion
      {% include "@components/usa-sidenav/src/usa-sidenav.twig" with sidenav_settings %}  
```

**Review by mejiaj — COMMENTED** [review]
Thanks for helping out with this. Small comment for clarity, otherwise LGTM.

File: packages/templates/usa-documentation/includes/_usa-docs-inner.twig (hunk 3 above)
Comment (mahoneycm) [reply packages/templates/usa-documentation/includes/_usa-docs-inner.twig:1]: I can dig it!

**Review by mejiaj — APPROVED** [review]

**Comment by mejiaj** [comment]
@mahoneycm I tweaked the summary to avoid any confusion with USA In-page navigation.

**Pull request merged**